app/screens/edit_profile/edit_profile.tsx

```diff
@@ -125,14 +128,25 @@ const EditProfile = ({
             if (!currentUser) {
                 return;
             }
-            const {error: fetchError, attributes} = await fetchCustomAttributes(serverUrl, currentUser.id);
-            if (!fetchError && attributes) {
-                setUserInfo((prev: UserInfo) => ({...prev, customAttributes: attributes} as UserInfo));
+
+            if (enableCustomAttributes) {
+                setUserInfo((prev) => ({
+                    ...prev,
+                    customAttributes: customAttributesSet || {},
+                }));
+            }
+
+            // Then fetch from server for latest data
+            const reqTime = Date.now();
+            lastRequest.current = reqTime;
+            const {error: fetchError, attributes} = await fetchCustomProfileAttributes(serverUrl, currentUser.id);
+            if (!fetchError && attributes && lastRequest.current === reqTime) {
```

Comment:
isn't lastRequest.current always the same as reqTime ?

Reply:
not if there are multiple requests, this was an improvement from @larkox in the original version without the database to prevent multiple requests overlapping